heartbeat/ocivip

```diff
@@ -0,0 +1,238 @@
+<action name="meta-data"    timeout="5s" />
+<action name="validate"     timeout="10s" />
+<action name="validate-all" timeout="10s" />
+</actions>
+</resource-agent>
+END
+}
+
+#######################################################################
+
+ocivip_usage() {
+    cat <<END
+usage: $0 {start|stop|monitor|migrate_to|migrate_from|validate|validate-all|meta-data}
+
+Expects to have a fully populated OCF RA-compliant environment set.
+END
+}
+
+ocivip_start() {
+    ocivip_monitor && return $OCF_SUCCESS
+
+    $OCICLI network vnic assign-private-ip --vnic-id $VNIC_ID \
+        --unassign-if-already-assigned \
+        --ip-address ${SECONDARY_PRIVATE_IP}
+    ip addr add ${SECONDARY_PRIVATE_IP}/24 dev ${PRIMARY_IFACE} label ${PRIMARY_IFACE}:0
+    RET=$?
+
+    # delay to avoid sending request too fast
+    sleep ${OCF_RESKEY_api_delay}
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+
+    ocf_log info "secondary_private_ip has been successfully brought up (${SECONDARY_PRIVATE_IP})"
+    return $OCF_SUCCESS
+}
+
+ocivip_stop() {
+    ocivip_monitor || return $OCF_SUCCESS
+
+    $OCICLI network vnic unassign-private-ip --vnic-id $VNIC_ID \
+        --ip-address ${SECONDARY_PRIVATE_IP}
+    ip addr del ${SECONDARY_PRIVATE_IP}/24 dev ${PRIMARY_IFACE}:0
+    RET=$?
+
+    # delay to avoid sending request too fast
+    sleep ${OCF_RESKEY_api_delay}
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+
+    ocf_log info "secondary_private_ip has been successfully brought down (${SECONDARY_PRIVATE_IP})"
+    return $OCF_SUCCESS
+}
+
+ocivip_monitor() {
+    $OCICLI network private-ip list --vnic-id $VNIC_ID | grep -q "${SECONDARY_PRIVATE_IP}"
+
+    RET=$?
+
+    if [ $RET -ne 0 ]; then
+        return $OCF_NOT_RUNNING
+    fi
+    return $OCF_SUCCESS
+}
+
+ocivip_validate() {
+    check_binary ${OCICLI}
+
+    if [ -z "${INSTANCE_ID}" ]; then
+        ocf_exit_reason "instance_id not found. Is this a Compute instance?"
+        return $OCF_ERR_GENERIC
+    fi
+
+    if [ -z "${VNIC_ID}" ]; then
+        ocf_exit_reason "vnic_id not found. Is this a Compute instance?"
+        return $OCF_ERR_GENERIC
+    fi
+
+    return $OCF_SUCCESS
+}
+
+case $__OCF_ACTION in
+    meta-data)
+        meta_data
+        exit $OCF_SUCCESS
+        ;;
+esac
+
+OCICLI="${OCF_RESKEY_ocicli}"
+SECONDARY_PRIVATE_IP="${OCF_RESKEY_secondary_private_ip}"
+INSTANCE_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.id')"
+VNIC_ID="$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/vnics/ | jq -r '.[0].vnicId')"
+COMPARTMENT_ID=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.compartmentId')
+AVAILABILITY_DOMAIN=$(curl -s -H "Authorization: Bearer Oracle" -L http://169.254.169.254/opc/v2/instance | jq -r '.availabilityDomain')
+PRIMARY_IFACE=$(ip -4 route ls | grep default | grep -Po '(?<=dev )(\S+)')
```

Comment:
It seems like you arent using all of these.

Reply:
Fixed [here](https://github.com/ClusterLabs/resource-agents/pull/1751/commits/5a5fefb1e8c5084c75335336cce4e5a7c43bec8a)